shell.js

```diff
@@ -114,8 +114,7 @@ rl.on('line', async function(cmd) {
 	} else {
 		// Get a reply from the bot.
 		if (bot && ready) {
-			var reply = await bot.reply("localuser", cmd);
-			console.log("Bot>", reply);
+			await bot.replyByBox("localuser", cmd).then(reply => reply.forEach(box => console.log("Bot>", box)));
```

Comment:
This line suggests the use case for this is rather limited. This line does a `console.log("Bot>", box)` for each part of the reply array but doesn't make it easy for user code to handle each reply with its own callback function.

RiveScript can be used in more than just console environments, and would need the callback function to (for example) send the reply over Slack or SMS or append it to an HTML document or so on.

So the other obstacle is that RiveScript doesn't return multiple replies, and in the Promise-based implementation it currently uses (`reply().then()`), it can't resolve multiple times to send multiple replies to the promise handler. What I would always recommend people do if they wanted multiple replies was for them to make up a `<nextreply>` tag and use it in their responses:

```
+ hello
- Hi there.<nextreply>How are you?
```

Then the reply() function would yield a literal `Hi there.<nextreply>How are you?` and the user code would split at the tag and deliver each reply in turn; but the core RiveScript API doesn't support yielding multiple replies itself.

---

It looks like the idea with the "boxed replies" is to send two sets of random replies in one response? If so, a workaround to do this using the vanilla rivescript-js as it currently stands could be like:

```
+ anything
- some<@int second reply>
- some one<@int second reply>

// "int"ernal trigger for the second part
+ int second reply
- any
- any one
```

Or just using `{random}` tags to avoid the need for extra replies:

```
+ anything
- {random}some|
^ some one{/random}
^ {random}any|
^ any one{/random}
```

Reply:
Thank you so much!!
I understand and agree with your suggestion.